File: LeetCode/Java/countLetters.java
/*
    1180. Count Substrings with Only One Distinct Letter
    https://leetcode.com/problems/count-substrings-with-only-one-distinct-letter/submissions/
 */

class Solution {
    public int countLetters(String S) {
        // int count = 0;
        // for (int i = 0; i < S.length(); i++) {
        //     count++;
        //     char temp = S.charAt(i);
        //     int j = i + 1;
        //     while (j < S.length() && temp == S.charAt(j)) {
        //         count++;
        //         j++;
        //     }
        // }
        // return count;
        
        int count = 1;
        int tempCount = 1;
        for (int i = 1; i < S.length(); i++) {
            if (S.charAt(i) == S.charAt(i - 1))
                tempCount++;
            else
                tempCount = 1;
            count += tempCount;
        }
        return count;
    }
}

/*
    Runtime: 0 ms, faster than 100.00% of Java online submissions for Count Substrings with Only One Distinct Letter.
    Memory Usage: 34.1 MB, less than 100.00% of Java online submissions for Count Substrings with Only One Distinct Letter.
 */

File: LeetCode/Swift/countLetters.swift
/*
    1180. Count Substrings with Only One Distinct Letter
    https://leetcode.com/problems/count-substrings-with-only-one-distinct-letter/submissions/
 */

class Solution {
    func countLetters(_ S: String) -> Int {
        var count = 1
        var tempCount = 1
        for n in 1..<S.count {
            if String(Array(S)[n]) == String(Array(S)[n-1]) {
                tempCount += 1
            } else {
                tempCount = 1
            }
            count += tempCount
        }
        return count;
    }
}

/*
    Runtime: 800 ms, faster than 6.67% of Swift online submissions for Count Substrings with Only One Distinct Letter.
    Memory Usage: 20.5 MB, less than 100.00% of Swift online submissions for Count Substrings with Only One Distinct Letter.
 */
